module/Company/src/Company/Mapper/Company.php

```diff
@@ -31,8 +31,61 @@ public function __construct(EntityManager $em)
     {
         $this->em = $em;
     }
+    /**
+     * Saves all unsaved entities, that are marked persistent
+     *
+     */
+    public function save()
+    {
+        $this->em->flush();
+    }
 
     /**
+     * Delete the company identified with $slug
+     *
+     * @param mixed $slug
+     */
+    public function deleteWithSlug($slug)
```

Comment:
Suppose you have three boxes, and one of them contains an apple. I can now throw away the box with the apple, or I can throw away the box by an apple. Clearly, the first is what is meant in this code.